Use explicit xarray's indexes: Delete UgridDataArray, UgridDataset

Huite opened the pull request:
With the changes so far, this snippet already works!

```python
import xarray as xr
import xugrid as xu

# %%
# Grab some data with UGRID conventions, run ugrid.initialize to create the UgridIndex:

ds = xu.data.elevation_nl(xarray=True)
uds = ds.ugrid.initialize()

# %%
# Grab one variable (would've been a "UgridDataArray" before) and try plotting it:
uda = uds["elevation"]
uda.ugrid.plot()

# %%
# Get it directly as a DataArray with a UgridIndex:
uda = xu.data.elevation_nl()
uda.ugrid.plot()
# %%

# Get rid of roughly half the data
masked = uda.where(uda["mesh2d_face_x"] < 150_000)
masked.ugrid.plot()
# %%

# Fill it back in with Laplace interpolation
filled = masked.ugrid.laplace_interpolate()
filled.ugrid.plot()
```

Commit:
Use explicit xarray's indexes: Delete UgridDataArray, UgridDataset

deltamarnix commented on xugrid/core/wrap.py:
```diff
@@ -1,655 +1,6 @@
-            stacklevel=2,
-        )
-        return UgridDataset.from_structured2d(dataset, topology)
+UgridDataArray = xr.DataArray
+UgridDataset = xr.Dataset
```
Does this automatically fix `isinstance(x, UgridDataset)` so we don't need to create classes with deprecation warnings?

deltamarnix commented on xugrid/core/index.py:
```diff
@@ -0,0 +1,38 @@
+        return True
+
+
+class UgridIndex1d(UgridIndex):
```
Are these classes part of the API? Perhaps add documentation and type hints

deltamarnix commented on xugrid/core/dataset_accessor.py:
```diff
@@ -4,18 +4,43 @@
+                node_ys[0],
+                ugrid_vars["face_node_connectivity"],
+            )
+            new = new.set_coords(variables).set_xindex(variables, UgridIndex2d)
```
How would you eventually envision going for [fully unstructured grids](https://ugrid-conventions.github.io/ugrid-conventions/#fully-3d-unstructured-ie-non-layered-mesh-topology) as a program like flopy would need?

deltamarnix reviewed (commented):
Amazing if we can pull this off and can get rid of custom classes.

Commit:
Merge branch 'main' into index

Commit:
More explicit index experimentation

Commit:
Add FutureWarnings

Commit:
Merge branch 'main' into index

Commit:
Explore xarray improvements to explicit indexes:

* Pin to xarray version >=2026.4.0; remove py3.10 and add py3.14.
* Replace deprecated UgridDataArray and UgridDataset constructors internally.
* Small start: update test_data tests.

Commit:
Make a start with updating tests for explicit UGRID indexes

Commit:
Merge branch 'main' into index

JoerivanEngelen committed:
Explicitly set compat

Huite committed:
Change merge compatibility to 'override' for dataset in ugrid1d

Commit:
WIP

Huite committed:
Update `pixi.lock`

veenstrajelmer committed:
Made some additional dependencies optional, local tests pass

dependabot[bot] committed:
Bump prefix-dev/setup-pixi from 0.9.5 to 0.9.6

Bumps [prefix-dev/setup-pixi](https://github.com/prefix-dev/setup-pixi) from 0.9.5 to 0.9.6.
- [Release notes](https://github.com/prefix-dev/setup-pixi/releases)
- [Commits](https://github.com/prefix-dev/setup-pixi/compare/v0.9.5...v0.9.6)

---
updated-dependencies:
- dependency-name: prefix-dev/setup-pixi
  dependency-version: 0.9.6
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

rsignell committed:
Update README to remove work in progress notice

Every package is a work in progress, right?   This reminds me of the early days of the web pages with the little guy digging...  😆 

I think this statement makes potential users question whether this will be useful to them, but I'm convinced that it likely will be!

Huite committed:
Update `pixi.lock`

dependabot[bot] committed:
Bump codecov/codecov-action from 6 to 7 (#443)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from 6 to 7.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v6...v7)

---
updated-dependencies:
- dependency-name: codecov/codecov-action
  dependency-version: '7'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

gaoflow committed:
Fix Ugrid2d.clip_box ignoring its bounding box arguments

The first line of Ugrid2d.clip_box reassigned xmin, ymin, xmax, ymax from
self.bounds, discarding the caller's arguments. As a result the method
always clipped to the grid's own full extent, i.e. it returned the whole
mesh regardless of the requested box.

Drop that line so the supplied bounds are used. (Ugrid1d.clip_box and
_sel_box already use the arguments correctly; _sel_box keeps self.bounds
only as a fallback for open-ended slices.)

Fixes #440.

Commit:
Update ci.yml to include test minimal dependency set

Commit:
Skip the convention tests that require netCDF4 via pooch -- should be
reworked!

Commit:
Fix edge indexing in nonperiodic conversion. Fixes #447

Commit:
Version 0.15.3

Commit:
Changelog typo

Commit:
Merge branch 'main' into index
